contracts/pcv/PCVDepositAggregator.sol

```diff
@@ -0,0 +1,365 @@
+
+        if (amount > totalBalance) {
+            revert("Not enough balance to withdraw");
+        }
+
+        // If our buffer is full enough, just transfer from that
+        if (IERC20(token).balanceOf(address(this)) >= amount) {
+            IERC20(token).safeTransfer(to, amount);
+        } else {
+            // We're going to have to pull from underlying deposits
+            // To avoid the need from a rebalance, we should only withdraw overages
+            // Calculate the amounts to withdraw from each underlying (this is basically half of a rebalance)
+
+            uint totalAmountNeeded = amount - IERC20(token).balanceOf(address(this));
+
+            for (uint i=0; i < pcvDepositAddresses.length(); i++) {
+                address pcvDepositAddress = pcvDepositAddresses.at(i);
+                uint128 pcvDepositWeight = pcvDepositInfos[pcvDepositAddress].weight;
+                uint actualPcvDepositBalance = IPCVDeposit(pcvDepositAddress).balance();
+                uint idealPcvDepositBalance = pcvDepositWeight * totalBalance / totalWeight;
+
+                // Only withdraw from this underlying if it has an overage 
+                if (actualPcvDepositBalance > idealPcvDepositBalance) {
+                    uint pcvDepositOverage = actualPcvDepositBalance - idealPcvDepositBalance;
+                    uint amountToWithdraw = pcvDepositOverage;
+                    
+                    if (totalAmountNeeded < pcvDepositOverage) {
+                        amountToWithdraw = totalAmountNeeded;
+                    }
+                    
+                    IPCVDeposit(pcvDepositAddress).withdraw(address(this), amountToWithdraw);
+                    totalAmountNeeded -= amountToWithdraw;
+
+                    // If we don't need to withdraw anymore, stop looping over the deposits
+                    if (totalAmountNeeded == 0) break;
+                } else {
+                    continue;
+                }
+            }
+
+            IERC20(token).safeTransfer(to, amount);
+        }
+    }
+
+    function withdrawERC20(address _token, address to, uint256 amount) external virtual override onlyPCVController {
+        IERC20(_token).safeTransfer(to, amount);
+    }
+
+    function withdrawETH(address payable to, uint256 amount) external virtual override onlyPCVController {
+        to.transfer(amount);
+    }
+
+    function setBufferWeight(uint128 newBufferWeight) external virtual override onlyGuardianOrGovernor {
+        int128 difference = int128(newBufferWeight) - int128(bufferWeight);
+        bufferWeight = uint128(int128(bufferWeight) + difference);
+
+        totalWeight = uint128(int128(totalWeight) + difference);
+    }
+
+    function setPCVDepositWeight(address depositAddress, uint128 newDepositWeight) external virtual override onlyGuardianOrGovernor {
+        if (!pcvDepositAddresses.contains(depositAddress)) {
+            revert("Deposit does not exist.");
+        }
```

Comment:
Why not add instead?

Reply:
Separation of responsibilities. setPCVDepositWeight should do one thing and one thing only: set an (existing) pcv deposit's weight.